packages/hardhat-core/src/builtin-tasks/flatten.ts

```diff
@@ -42,7 +42,11 @@ function getSortedFiles(dependenciesGraph: DependencyGraph) {
     const sortedNames = [...new Set(withEntries)];
     return sortedNames.map((n) => filesMap[n]);
   } catch (error) {
-    if (error.toString().includes("Error: There is a cycle in the graph.")) {
+    if (
```

Comment:
I vote yes.

Reply:
Sure, let's upgrade it! 🚀

Reply:
When doing the assertion I get the error "Property 'message' does not exist on type 'object'", can I just use any instead of unknown? This is after updating to prettier 2.1.2 not that that should matter

```
    } catch (error: unknown) {
      assertHardhatInvariant(
        typeof error === "object" && error !== null,
        "Error is not an object"
      );
      if (typeof error.message === "string" &&
        error.message.includes("no matching file exists") //error occurs here
      ) 
```